src/wallet/wallet.cpp

```diff
@@ -2946,73 +2946,31 @@ void CWallet::AvailableCoins(vector<COutput>& vCoins, bool fOnlyConfirmed, const
                 }
-
-            }
-        }
-    }
-}
-
-static void ApproximateBestSubset(
-    vector<pair<CAmount, pair<const CWalletTransactionBase*,unsigned int> > >vValue, const CAmount& nTotalLower, const CAmount& nTargetValue,
-    vector<char>& vfBest, CAmount& nBest, int iterations = 1000)
-{
-    vector<char> vfIncluded;
-
-    vfBest.assign(vValue.size(), true);
-    nBest = nTotalLower;
-
-    seed_insecure_rand();
-
-    for (int nRep = 0; nRep < iterations && nBest != nTargetValue; nRep++)
-    {
-        vfIncluded.assign(vValue.size(), false);
-        CAmount nTotal = 0;
-        bool fReachedTarget = false;
-        for (int nPass = 0; nPass < 2 && !fReachedTarget; nPass++)
-        {
-            for (unsigned int i = 0; i < vValue.size(); i++)
-            {
-                //The solver here uses a randomized algorithm,
-                //the randomness serves no real security purpose but is just
-                //needed to prevent degenerate behavior and it is important
-                //that the rng is fast. We do not use a constant random sequence,
-                //because there may be some privacy improvement by making
-                //the selection random.
-                if (nPass == 0 ? insecure_rand()&1 : !vfIncluded[i])
-                {
-                    nTotal += vValue[i].first;
-                    vfIncluded[i] = true;
-                    if (nTotal >= nTargetValue)
-                    {
-                        fReachedTarget = true;
-                        if (nTotal < nBest)
-                        {
-                            nBest = nTotal;
-                            vfBest = vfIncluded;
-                        }
-                        nTotal -= vValue[i].first;
-                        vfIncluded[i] = false;
-                    }
-                }
             }
         }
     }
 }
 
 bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, int nConfMine, int nConfTheirs, vector<COutput> vCoins,
-                                 set<pair<const CWalletTransactionBase*,unsigned int> >& setCoinsRet, CAmount& nValueRet) const
+                                 vector<COutput>& vCoinsRet, CAmount& nValueRet, size_t &selectionTotalBytes,
+                                 size_t availableBytes, bool useInputsNetValues) const
 {
-    setCoinsRet.clear();
+    vCoinsRet.clear();
     nValueRet = 0;
+    selectionTotalBytes = 0;
 
     // List of values less than target
-    pair<CAmount, pair<const CWalletTransactionBase*,unsigned int> > coinLowestLarger;
-    coinLowestLarger.first = std::numeric_limits<CAmount>::max();
-    coinLowestLarger.second.first = NULL;
-    vector<pair<CAmount, pair<const CWalletTransactionBase*,unsigned int> > > vValue;
-    CAmount nTotalLower = 0;
+    // <gross value, value, output, size>
+    vector<tuple<CAmount, CAmount, COutput, size_t>> vValue;
```

Comment:
Either use explicitly `std::` namespace or don't but be consistent (see below std::numeric_limits)
Preferrable to use it explicitly

Reply:
I agree with your suggestion.
Modified accordingly (refactored all `vector`, `string`, `set`, `map`, `pair`, `tuple`, `make_pair`, `make_tuple` to _std::X_ version inside `wallet.cpp`).